### PR TITLE
Update Deploy Seq2Seq Tutorial with New TorchScript API

### PR DESCRIPTION
**Pull request by wanchaol**

### CODE REVIEW

File: beginner_source/deploy_seq2seq_hybrid_frontend_tutorial.py [hunk 1]
```diff
@@ -1,24 +1,24 @@
...
+# `Chatbot tutorial <https://pytorch.org/tutorials/beginner/chatbot_tutorial.html>`__.
 # You can either treat this tutorial as a “Part 2” to the Chatbot tutorial
 # and deploy your own pretrained model, or you can start with this
 # document and use a pretrained model that we host. In the latter case,
 # you can reference the original Chatbot tutorial for details
 # regarding data preprocessing, model theory and definition, and model
 # training.
 #
-# What is the Hybrid Frontend?
+# What is the TorchScript?
```
Comment (Chillee) [inline beginner_source/deploy_seq2seq_hybrid_frontend_tutorial.py]: What is Torchscript?

File: beginner_source/deploy_seq2seq_hybrid_frontend_tutorial.py [hunk 2]
```diff
@@ -52,23 +52,20 @@
...
-# reference <https://pytorch.org/docs/master/jit.html>`__. To provide the
-# maximum flexibility, the modes of Torch Script can be composed to
-# represent your whole program, and these techniques can be applied
-# incrementally.
-#
-# .. figure:: /_static/img/chatbot/pytorch_workflow.png
-#    :align: center
-#    :alt: workflow
+# control flow, a **scripting** mechanism is provided. The
+# ``torch.jit.script`` function takes module or function and does not
```
Comment (Chillee) [inline beginner_source/deploy_seq2seq_hybrid_frontend_tutorial.py]: The ``torch.jit.script`` decorator takes a module or function...

File: beginner_source/deploy_seq2seq_hybrid_frontend_tutorial.py [hunk 3]
```diff
@@ -52,23 +52,20 @@
...
+# requires example inputs. Scripting then explicitly converts the module
+# or function code to TorchScript, including all possible control flow
+# routes. The one caveat with using scripting is that it only supports
```
Comment (Chillee) [inline beginner_source/deploy_seq2seq_hybrid_frontend_tutorial.py]: probably "all control flow" instead of "all possible control flow routes"?

Comment (Chillee) [reply beginner_source/deploy_seq2seq_hybrid_frontend_tutorial.py]: Also, I think "One caveat with using scripting" is a bit more natural than "The one caveat..."

File: beginner_source/deploy_seq2seq_hybrid_frontend_tutorial.py [hunk 4]
```diff
@@ -383,14 +381,14 @@ def forward(self, hidden, encoder_outputs):
...
 # attention to. From here, we use a linear layer and softmax normalization
 # to select the next word in the output sequence.
-#
-# Hybrid Frontend Notes:
+
+# TorchScript Notes:
 # ~~~~~~~~~~~~~~~~~~~~~~
 #
-# Similarly to the ``EncoderRNN``, this module does not contain any
+# Similarly to the ``EncoderRNN```, this module does not contain any
```
Comment (Chillee) [inline beginner_source/deploy_seq2seq_hybrid_frontend_tutorial.py]: Is the extra ` on purpose?

File: beginner_source/deploy_seq2seq_hybrid_frontend_tutorial.py (hunk 2 above)
Comment (wanchaol) [reply beginner_source/deploy_seq2seq_hybrid_frontend_tutorial.py]: `torch.jit.script` right now does not necessarily to be a decorator right?

Comment (wanchaol) [reply beginner_source/deploy_seq2seq_hybrid_frontend_tutorial.py]: like you can call it `scripted_model = torch.jit.script(model)`, which is not a decorator.

File: beginner_source/deploy_seq2seq_hybrid_frontend_tutorial.py [hunk 5]
```diff
@@ -1,24 +1,24 @@
 # -*- coding: utf-8 -*-
 """
-Deploying a Seq2Seq Model with the Hybrid Frontend
+Deploying a Seq2Seq Model with the TorchScript
```
Comment (Chillee) [inline beginner_source/deploy_seq2seq_hybrid_frontend_tutorial.py]: with ~the~ Torchscript

**Comment by suo** [comment]
This looks good to me, but needs a rebase @wanchaol

**Comment by wanchaol** [comment]
close this in favor of https://github.com/pytorch/tutorials/pull/597

**Pull request closed without merging**